File: LeetCode/Simulation/Python/Leetcode682-棒球比赛.py
class Solution:
    def calPoints(self, ops: List[str]) -> int:
        rec,ans = [],0
        for op in ops:
            if op[0] == 'D':
                ans += rec[-1] * 2
                rec.append(rec[-1] * 2)
            elif op[0] =='+':
                ans += rec[-1] + rec[-2]
                rec.append(rec[-1] + rec[-2])
            elif op[0] == 'C':
                ans -= rec[-1]
                rec.pop()
            else:
                ans += int(op)
                rec.append(int(op))
        return ans

File: LeetCode/Simulation/JS/Leetcode682-棒球比赛.js
var calPoints = function(ops) {
    const rec = [];
    let ans = 0;
    for(const op of ops) {
        const n = rec.length;
        switch(op[0]) {
            case 'D':
                ans += rec[n - 1] * 2;
                rec.push(rec[n - 1] * 2);
                break;
            case 'C':
                ans -= rec[n - 1];
                rec.pop();
                break;
            case '+':
                ans += rec[n - 1] + rec[n - 2];
                rec.push(rec[n - 1] + rec[n - 2]);
                break;
            default:
                ans += parseInt(op);
                rec.push(parseInt(op));
                break;
        }
    }
    return ans;
};